errors.go

```diff
@@ -55,16 +57,15 @@ func isResponseError(req *http.Request, resp *http.Response) *ResponseError {
 		return nil
 	}
 
-	// 429 is returned by standby instances for /sys/health requests and should
-	// not be treated as an error; for other paths the status code indicates
-	// that the quota limit has been reached.
-	if resp.StatusCode == http.StatusTooManyRequests /* 429 */ && req.URL.Path == "/v1/sys/health" {
-		return nil
-	}
-
-	// 472 & 473 are returned by (performance/secondary) standby instances for /sys/health requests and should
-	// not be treated as an error
-	if (resp.StatusCode == 472 || resp.StatusCode == 473) && req.URL.Path == "/v1/sys/health" {
+	// /v1/sys/health returns a few special 4xx status codes that should not be
+	// treated as errors:
+	//
+	//  - 429 if unsealed and standby
+	//  - 472 if disaster recovery mode replication secondary and active
+	//  - 473 if performance standby
+	//
+	// See: https://developer.hashicorp.com/vault/api-docs/system/health
+	if req.URL.Path == "/v1/sys/health" && slices.Contains([]int{429, 472, 473}, resp.StatusCode) {
 		return nil
```

Comment:
I see the library already has a dependency on `x/exp` for just one other case of evaluating HTTP status codes. Rather than furthering it, how about if we cut out the `x/exp` dependency entirely, to reduce the dependency footprint?

There is a fairly concise way to write it with only core Go syntax (and it probably saves an allocation, too):

```suggestion
	if req.URL.Path == "/v1/sys/health" {
		switch resp.StatusCode {
		case 429, 472, 473:
			return nil
		}
	}
```

Reply:
I'm not too concerned about including x/exp since it will likely be part of the standard library in the near future, but I do like this approach, thanks!